packages/react/id/src/id.tsx

```diff
@@ -1,45 +1,16 @@
-// This implementation is heavily inspired by react-aria's implementation
-// See: https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/ssr/src/SSRProvider.tsx
-
 import * as React from 'react';
-import { createContext } from '@radix-ui/react-context';
-
-const PROVIDER_NAME = 'IdProvider';
-
-const defaultIdContext = {
-  prefix: Math.round(Math.random() * 10000000000),
-  current: 0,
-};
+import { useLayoutEffect } from '@radix-ui/react-use-layout-effect';
 
-const [IdProviderImpl, useIdContext] = createContext(PROVIDER_NAME, defaultIdContext);
-
-const IdProvider: React.FC<{ children: React.ReactNode }> = (props) => {
-  const currentContext = useIdContext(PROVIDER_NAME);
-  const isRootIdProvider = currentContext === defaultIdContext;
-  return (
-    <IdProviderImpl
-      prefix={isRootIdProvider ? 0 : ++currentContext.prefix}
-      current={0}
-      {...props}
-    />
-  );
-};
+const useReactId = (React as any).useId || (() => undefined);
```

Comment:
Do not use `(React as any).useId`, it's break project.

Reply:
Popular bundlers try to import useId from React

Reply:
#1023

Reply:
This has been fixed here https://github.com/radix-ui/primitives/pull/1028